cryptographic_estimators/UOVEstimator/UOVAlgorithms/direct_attack.py

```diff
@@ -0,0 +1,160 @@
+                                        memory_access=0,
+                                        complexity_type=complexity_type,
+                                        bit_complexities=0)
+        self._fastest_algorithm = None
+
+    def get_fastest_mq_algorithm(self):
+        if self._fastest_algorithm is None:
+            self._fastest_algorithm = self._MQEstimator.fastest_algorithm()
+        return self._fastest_algorithm
+
+    def expected_number_solutions(self):
+        """
+        Returns the logarithm of the expected number of existing solutions to the problem
+        """
+        n, m, q = self.problem.get_parameters()
+        return log2(q) * (n - m)
+
+    def _compute_time_complexity(self, parameters: dict):
+        """
+        Return the time complexity of the algorithm for a given set of parameters
+
+        INPUT:
+
+        - ``parameters`` -- dictionary including the parameters
+
+        TESTS::
+
+            sage: from cryptographic_estimators.UOVEstimator.UOVAlgorithms.direct_attack import DirectAttack
+            sage: from cryptographic_estimators.UOVEstimator.uov_problem import UOVProblem
+            sage: A = DirectAttack(UOVProblem(n=14, m=12, q=5))
+            sage: A.time_complexity()
+            29.54708305178623
+
+        """
+        fastest_algorithm = self.get_fastest_mq_algorithm()
+        fastest_algorithm.complexity_type = self.complexity_type
+        return self._fastest_algorithm.time_complexity()
+
+    def _compute_memory_complexity(self, parameters: dict):
+        """
+        Return the memory complexity of the algorithm for a given set of parameters
+
+        INPUT:
+
+        - ``parameters`` -- dictionary including the parameters
+
+        TESTS::
+
+            sage: from cryptographic_estimators.UOVEstimator.UOVAlgorithms.direct_attack import DirectAttack
+            sage: from cryptographic_estimators.UOVEstimator.uov_problem import UOVProblem
+            sage: A = DirectAttack(UOVProblem(n=14, m=12, q=5))
+            sage: A.memory_complexity()
+            16.13937148392593
+
+        """
+        fastest_algorithm = self.get_fastest_mq_algorithm()
+        fastest_algorithm.complexity_type = self.complexity_type
+        return self._fastest_algorithm.memory_complexity()
+    
+    def _compute_tilde_o_time_complexity(self, parameters: dict):
+        """
+        Return the Ō time complexity of the algorithm for a given set of parameters
+
+        INPUT:
+
+        - ``parameters`` -- dictionary including the parameters
+
+        TESTS::
+
+            sage: from cryptographic_estimators.UOVEstimator.UOVAlgorithms.direct_attack import DirectAttack
+            sage: from cryptographic_estimators.UOVEstimator.uov_problem import UOVProblem
+            sage: A = DirectAttack(UOVProblem(n=14, m=12, q=5), complexity_type=1)
+            sage: A.time_complexity()
+            23.520125380651073
+        """
+        fastest_algorithm = self.get_fastest_mq_algorithm()
```

Comment:
only something minor/odd. 
` _compute_tilde_o_time_complexity` and `_compute_time_complexity` are the same function. They only differ in the context they are called. So theoretically it would be possible to set the estimator to `complexity_type=0` (estimate), call ` _compute_tilde_o_time_complexity` and still get the normal estimation number and not the asymptotic one.
Wouldn`t it be better to hardcode the two different complexity types values within the function itself?